starlite/handlers/http.py

```diff
@@ -225,6 +233,23 @@ def resolve_after_request(self) -> Optional[AfterRequestHandler]:
                 self.resolved_after_request = self.resolved_after_request.__func__
         return cast(Optional[AfterRequestHandler], self.resolved_after_request)
 
+    def resolve_after_response(self) -> Optional[AfterResponseHandler]:
+        """
+        Resolves the after_response handler by starting from the route handler and moving up.
+
+        If a handler is found it is returned, otherwise None is set.
+        This method is memoized so the computation occurs only once
+        """
+        if self.resolved_after_response is BaseRouteHandler.empty:
+            self.resolved_after_response = None
+            for layer in self.ownership_layers:
+                if layer.after_response:
+                    self.resolved_after_response = layer.after_response
+            if self.resolved_after_response is not None and ismethod(self.resolved_after_response):
+                # python automatically binds class variables, which we do not want in this case.
+                self.resolved_after_response = self.resolved_after_response.__func__
```

Comment:
Does using 
```python
if layer.after_response:
    self.resolved_after_response = staticmethod(layer.after_response)
```
remove the need to do this?

Reply:
I didn't have a lot of time when reviewing, but I had several mypy errors when I quickly tested this.

Reply:
I get 1:

```python
starlite/handlers/http.py:247: error: Incompatible types in assignment (expression has type "staticmethod[Optional[Awaitable[None]]]", variable has type "Union[Callable[[Request[Any, Any]], None], Callable[[Request[Any, Any]], Awaitable[None]], None, Type[empty]]")  [assignment]
```
Which is a result of the type that we declare for the attribute in `__init__()`. Just to be clear, I don't think this is a big deal, but just asking the question as `staticmethod()` is kinda the front-door to avoiding implicit binding of functions to class instances, from [the docs](https://docs.python.org/3/library/functions.html#staticmethod):

> This is needed in some cases where you need a reference to a function from a class body and you want to avoid the automatic transformation to instance method

Reply:
So static method causes test failures. You are welcome to look into it if and when you have the time.

Reply:
yep that's cool - it would prob. be better to use everywhere or nowhere for consistencies sake anyway :+1: